collections/infrastructure/roles/clustershell/tasks/main.yml

```diff
@@ -33,3 +33,26 @@
     mode: 0644
   tags:
     - template
+  when: clustershell_custom_map == false
```

Comment:
Why using clustershell_custom_map variable ?
In the current logic, you could set that if "clush" (to be renamed) variable is set, then local.cfg is not generated. Else, it is generated.
However, I think both should be able to work together.
I need to recheck clustershell configuration, but would it work with both local.cfg and cluster.cfg files ?
If yes, then maybe a solution would be to just check if "clush" is defined, and then generate cluster.cfg, and always generate local.cfg. Or if you think local.cfg could somehow "spam" clustershell configuration, indeed create a variable in defaults like "clustershell_inventory_groups" to manipulate local.cfg generation.
What do you think ? 😊

Reply:
Good idea, I can just check for inventory existence.

In the model I created, it creates both. It always generates local.cfg and cluster.yaml.

The difference between these files is that local.cfg has a structure of: @name:nodelist.

Cluster.yaml allows you to create what they call 'roles' for each set of nodes. For example for Lustre, you could do:

@io:mds and @io:oss in cluster.yaml

And @io in local.cfg.

The templates in the modifications I made either generate the local.cfg based on the Ansible inventory or generate the local.cfg and the cluster.yaml based on the mapping that was defined in the clush.yml (to be changed), but the local.cfg is always created.

I will now work on those improvements that we commented here. I had an insight to improve these templates as well.